src/common/reducers/entities/index.js

```diff
@@ -26,9 +28,28 @@ export function entities(state = {
     return reduceRepoEntity(state, action);
   }
 
+  // update snaps on register name actions
+  if (RegisterNameActionTypes[action.type]) {
+    // TODO refactor
+    // register name actions use fullName as id instead of repo url
+    const snapAction = {
+      ...action,
+      payload: {
+        ...action.payload,
+        id: getGitHubRepoUrl(action.payload.id)
```

Comment:
Actions from `register-name` use repo full name as an id, while we use url as snap in store.
So before it's changed, we need to transform one id into another.